contracts/gateway/EIP20Gateway.sol

```diff
@@ -617,21 +615,21 @@ contract EIP20Gateway is GatewayBase {
         staker_ = message.sender;
         stakerNonce_ = message.nonce;
         amount_ = stakes[_messageHash].amount;
+        uint256 stakeBounty = stakes[_messageHash].bounty;
 
-        // transfer the staked amount to the staker
+        delete stakes[_messageHash];
+
+        // Transfer the staked amount to the staker.
         token.transfer(message.sender, amount_);
 
-        // burn facilitator bounty
-        baseToken.transfer(burner, bounty);
```

Comment:
This change is hard for me to follow. Where was the original `bounty` taken from??

Reply:
I cannot find the variable `bounty` in the original code 😳